### PR TITLE
refactor(pipeline): add processedTasks syncmap to determine normally done

### PR DESCRIPTION
**Pull request by sfwn**
#### What this PR does / why we need it:

refacotor pipeline reconcile loop, add processedTasks syncmap to determine normally done.

#### Specified Reviewers:

/assign @chengjoey 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  add processedTasks syncmap to determine normally done            |
| 🇨🇳 中文    |  增加 已处理的任务 字段来标识正常的推进结束            |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/providers/reconciler/pipeline_reconciler.go [hunk 1]
```diff
@@ -89,12 +87,25 @@ type defaultPipelineReconciler struct {
 	// canceling
 	flagCanceling bool
 
-	// flag have task
-	flagHaveTask *bool
+	// task related
+	totalTaskNumber            *int
```
Comment (chengjoey) [inline modules/pipeline/providers/reconciler/pipeline_reconciler.go:91]: maybe it's unnecessary to set `totalTaskNumber` pointer type, there are not found `totalTaskNumber` compare with nil

File: modules/pipeline/providers/reconciler/reconcile.go [hunk 2]
```diff
@@ -65,29 +65,28 @@ func (r *provider) ReconcileOnePipeline(ctx context.Context, pipelineID uint64)
...
 	return pr
 }
 
-func (pr *defaultPipelineReconciler) releaseTasksCanBeConcurrentlyScheduled(ctx context.Context, p *spec.Pipeline, tasks []*spec.PipelineTask) {
-	for _, task := range tasks {
-		pr.processingTasks.Delete(task.NodeName())
-	}
+func (pr *defaultPipelineReconciler) releaseTaskAfterReconciled(ctx context.Context, p *spec.Pipeline, task *spec.PipelineTask) {
+	pr.processingTasks.Delete(task.NodeName())
+	pr.processedTasks.Store(task.Name, struct{}{})
```
Comment (chengjoey) [inline modules/pipeline/providers/reconciler/reconcile.go:89]: use task.NodeName()

File: modules/pipeline/providers/reconciler/pipeline_reconciler.go (hunk 1 above)
Comment (sfwn) [reply modules/pipeline/providers/reconciler/pipeline_reconciler.go:91]: pointer type is more clear to distinguish from truly zero task.

File: modules/pipeline/providers/reconciler/reconcile.go (hunk 2 above)
Comment (sfwn) [reply modules/pipeline/providers/reconciler/reconcile.go:89]: yes, I will adjust it.

**Review by chengjoey — APPROVED** [review]

**Pull request merged**